cpp/src/parquet/encoding.cc

```diff
@@ -2482,7 +2482,7 @@ struct ByteStreamSplitDecodePath<T, true> {
   static int Decode(T* buffer, int max_values, int num_values_in_buffer,
                     const uint8_t* data, int& num_values_decoded, int& len_remaining) {
     constexpr size_t num_streams = sizeof(T);
-    static_assert(num_streams == 4U || num_streams == 8U);
+    static_assert(num_streams == 4U || num_streams == 8U, "Invalid number of streams.");
```

Comment:
Or: "Unsupported type for BYTE_STREAM_SPLIT decoding." ("Invalid number of streams." is just the consequence via sizeof(T))